### PR TITLE
DEVEXP-464:  Update Verification API

### PR DESCRIPTION
**Pull request by 650elx**
https://tickets.sinch.com/browse/DEVEXP-464

### CODE REVIEW

**Review by JPPortier — CHANGES_REQUESTED** [review]
Outside of implementation details, there is a lot of breaking changes but not highlighted by deprecation nor major version change.
The 'main' branch target do not fit with our branching strategy.
Let' talk about these points

File: sinch/domains/verification/models/requests.py [hunk 1]
```diff
@@ -4,6 +4,42 @@
 from sinch.domains.verification.models import VerificationIdentity
 
 
+class ReporPhoneCallVerificationDataTransfromationMixin:
```
Comment (asein-sinch) [inline sinch/domains/verification/models/requests.py:7]: Check for typos:

1. ReporPhoneCallVerificationDataTransfromationMixin
2. ReporFlashCallVerificationDataTransfromationMixin
3. ReporSMSVerificationDataTransfromationMixin

File: sinch/domains/verification/__init__.py [hunk 2]
```diff
@@ -156,16 +211,11 @@ def get_by_reference(self, reference) -> GetVerificationStatusByReferenceRespons
             )
         )
 
-    def get_by_id(self, id) -> GetVerificationStatusByIdResponse:
```
Comment (asein-sinch) [inline sinch/domains/verification/__init__.py:159]: I'm not sure to understand why this endpoint has been removed (moreover without any deprecation warning as mentioned by @JPPortier), it still exists in the specification: https://gitlab.com/sinch/sinch-projects/enterprise-and-messaging/documentation/developer-experience/oas-documentation/-/blob/master/services/verification-oas/verification.yaml?ref_type=heads#L105

**Comment by 650elx** [comment]
> Outside of implementation details, there is a lot of breaking changes but not highlighted by deprecation nor major version change. The 'main' branch target do not fit with our branching strategy. Let' talk about these points

Indeed. I'm going to bump the major version on the next release.

File: sinch/domains/verification/models/requests.py (hunk 1 above)
Comment (650elx) [reply sinch/domains/verification/models/requests.py]: indeed; copy/paste typos
fixed

File: sinch/domains/verification/__init__.py (hunk 2 above)
Comment (650elx) [reply sinch/domains/verification/__init__.py:159]: Oops! I've misunderstood the spec file. The endpoint is back.

**Comment by 650elx** [comment]
Closing due to #36

**Pull request closed without merging**